### DISCUSSION TITLE
can't read small caps

### DISCUSSION THREAD

**Discussion by afriedman412** [jsvine/pdfplumber#851] [Ask for help with specific PDFs]
Hi --
Some of the text in the attached PDF is in small caps and pdfplumber apparently can't read those characters.

This is example data from one of the characters...
```
{'matrix': (0.74999996875,
  0.0,
  0.0,
  0.74999996875,
  147.22417261565923,
  685.5000044375),
 'fontname': 'Georgia-Bold',
 'adv': 19.6328132,
 'upright': True,
 'x0': 147.22417261565923,
 'y0': 680.8960007493336,
 'x1': 161.94878190213382,
 'y1': 701.8959998743335,
 'width': 14.724609286474589,
 'height': 20.99999912499993,
 'size': 20.99999912499993,
 'object_type': 'char',
 'page_number': 1,
 'text': '\x00',
 'stroking_color': None,
 'non_stroking_color': None,
 'top': 90.10400012566652,
 'bottom': 111.10399925066645,
 'doctop': 90.10400012566652}
```
You can see it's interpreting it as `\x00`, and it comes up as nothing when I import the text. For example, "Filer Information" just comes through as "F I".

Any ideas?

[temp.pdf](https://github.com/jsvine/pdfplumber/files/11142365/temp.pdf)

**Comment by jsvine** [comment]
Thanks for sharing this example @afriedman412. It's a real stumper! I've spent some time now trying to track down what's happening. One initial, general observation: This problem doesn't seems specific to `pdfplumber`. If you try copying the text directly from the PDF (in whatever browser or PDF viewer you prefer), I believe you'll _also_ not pick up the small caps' text when you paste it into a text editor. (Or does it paste successfully for you?)

That said, I was still curious to know what was going on, and to see if I could track down the issue. I'm not 100% confident, but I think it may have something to do with this portion of the raw PDF, within the definition of the "Georgia-Bold" font (via [this tool](https://brendandahl.github.io/pdf.js.utils/browser/)):

<img width="475" alt="Screen Shot" src="https://user-images.githubusercontent.com/534702/230665267-22aeebfd-ebe3-44f9-9d09-d40e5f4081bb.png">

The `beginbfchar` and `endbfchar` commands (of which I was previously unfamiliar) seem to remap certain font glyphs to specific unicode characters; and above you see a bunch of them being remapped to what seems to be those `\x00`s that are showing up.

Unfortunately, I'm not aware of a simple/direct way of reversing those remappings.
